astroid/cache.py

```diff
@@ -0,0 +1,104 @@
+""" Caching utilities used in various places."""
+
+import typing
+from collections import OrderedDict
+from weakref import WeakSet
+
+import wrapt
+
+LRU_CACHE_CAPACITY = 128
```

Comment:
```suggestion
LRU_CACHE_CAPACITY = 2**30
```

Reading the test it seems it's a number of entry in the cache and not the sizeof taken in Mb of RAM. I suppose calculating the size in ram would take performance away ? Or is it impossible ? This is counter intuitive to me as I expected the cache capacity to be the space taken  in RAM. What do you think?

Reply:
I agree it's a bit counter-intuitive, but as far as I know, there is no efficient way to calculate the memory footprint of objects.
 
[`sys.getsizeof(obj)`](https://docs.python.org/3/library/sys.html#sys.getsizeof) is an option, but it only returns the size of the object itself and does not include the size of the objects that are referenced. So if we want to calculate the total size of an AST, we would need to traverse the entire tree and accumulate the size of the tree nodes. I believe this would be pretty expensive.

Reply:
I think taking only the size of the current node into account is fine. We're not really storing the whole ast in the cache, only this node, right ? The node permit to navigate to other part of the tree if we want, but there's no additional cost for it. 

Some random ideas : we could check the way `functool.lru_cache` handle this ? Because there is a maxsize when using it. Also maybe it's possible to use it directly, I'm no expert in caching but I suppose there is a way to reuse the existing and override it.

Reply:
> We're not really storing the whole ast in the cache, only this node, right ? The node permit to navigate to other part of the tree if we want, but there's no additional cost for it.

I'm not 100% sure, but it seems AST nodes could reference other AST nodes via various fields. Anyway, AST nodes are just an example. Any object that is referenced via a field is not accounted by `sys.getsizeof()`.

> Some random ideas : we could check the way functool.lru_cache handle this ? Because there is a maxsize when using it. 

`maxsize` limits the number of entries in the cache, not the memory footprint. Please see:
https://github.com/python/cpython/blob/8d7644fa64213207b8dc6f555cb8a02bfabeced2/Lib/functools.py#L479

The main reason I couldn't reuse `functool.lru_cache` is because it hold the cache as a closure variable, and thus there is no way to clear it from outside.

Reply:
Ho ok, it's in fact exactly the same than functools.lru_cache, thanks for correcting me.